src/controllers/mainController.ts

```diff
@@ -443,7 +443,19 @@ export default class MainController implements vscode.Disposable {
 
 		// Generate Azure Function command
 		this._context.subscriptions.push(vscode.commands.registerCommand(Constants.cmdCreateAzureFunction, async (node: TreeNodeInfo) => {
-			const connectionDetails = ConnectionCredentials.createConnectionDetails(node.connectionInfo);
+			let connectionInfo = node.connectionInfo;
+			// set the database containing the selected table so it can be used
+			// for the initial catalog property of the connection string
+			let newNode: TreeNodeInfo = node;
+			while (true) {
```

Comment:
`while(newNode)` - if for some reason there isn't a db node then we should still continue on and use the default connection db.